modules/reactor/src/meshgenerators/PinMeshGenerator.C

```diff
@@ -0,0 +1,480 @@
+
+      addMeshSubgenerator("PolygonConcentricCircleMeshGenerator", name() + "_circle", params);
+    }
+
+    // Rotate pin to be square rather than diamond, like what is generated using
+    // ConcentricCircleMeshGenerator
+    {
+      auto params = _app.getFactory().getValidParams("TransformGenerator");
+      params.set<MeshGeneratorName>("input") = name() + "_circle";
+      params.set<MooseEnum>("transform") = 4;
+      params.set<RealVectorValue>("vector_value") = RealVectorValue(0, 0, 45);
+
+      addMeshSubgenerator("TransformGenerator", name() + "trans", params);
+    }
+
+    // Pass mesh meta-data from subgenerator to this MeshGenerator
+    if (hasMeshProperty("pitch_meta", name() + "_circle"))
+      declareMeshProperty("pitch_meta", getMeshProperty<Real>("pitch_meta", name() + "_circle"));
+    if (hasMeshProperty("background_intervals_meta", name() + "_circle"))
+      declareMeshProperty(
+          "background_intervals_meta",
+          getMeshProperty<unsigned int>("background_intervals_meta", name() + "_circle"));
+    if (hasMeshProperty("node_id_background_meta", name() + "_circle"))
+      declareMeshProperty(
+          "node_id_background_meta",
+          getMeshProperty<unsigned int>("node_id_background_meta", name() + "_circle"));
+    if (hasMeshProperty("pattern_pitch_meta", name() + "_circle"))
+      declareMeshProperty("pattern_pitch_meta",
+                          getMeshProperty<Real>("pattern_pitch_meta", name() + "_circle"));
+    if (hasMeshProperty("azimuthal_angle_meta", name() + "_circle"))
+      declareMeshProperty(
+          "azimuthal_angle_meta",
+          getMeshProperty<std::vector<Real>>("azimuthal_angle_meta", name() + "_circle"));
+    if (hasMeshProperty("num_sectors_per_side_meta", name() + "_circle"))
+      declareMeshProperty("num_sectors_per_side_meta",
+                          getMeshProperty<std::vector<unsigned int>>("num_sectors_per_side_meta",
+                                                                     name() + "_circle"));
+    if (hasMeshProperty("max_radius_meta", name() + "_circle"))
+      declareMeshProperty("max_radius_meta",
+                          getMeshProperty<Real>("max_radius_meta", name() + "_circle"));
+
+    // Define boundary IDs so that there are unified boundaries for use in patterned MeshGenerator
+    // if this pin is to be used in an AssemblyMeshGenerator
+    {
+      auto params = _app.getFactory().getValidParams("SideSetsFromNormalsGenerator");
+
+      params.set<MeshGeneratorName>("input") = name() + "trans";
+      params.set<std::vector<Point>>("normals") = {
+          {1, 0, 0},
+          {-1, 0, 0},
+          {0, 1, 0},
+          {0, -1, 0}}; // normal directions over which to define boundaries
+      params.set<bool>("fixed_normal") = true;
+      params.set<bool>("replace") = false;
+      params.set<std::vector<BoundaryName>>("new_boundary") = {"10001", "10002", "10003", "10004"};
```

Comment:
10001 and 10003 seem wrong in the case I ran
![image](https://user-images.githubusercontent.com/25037151/156302002-1e4552a4-9e72-481f-a13c-1b132c24c506.png)

Maybe the orientation of the sideset has an issue in PolygonConcentricCircleMeshGenerator ?

Reply:
Once again, this is a deeper issue in how boundary id's are assigned for the Cartesian case, where we need to define many intermediate sidesets to get to the final sideset name of `outer_pin_1`. Ideally, we'd be defining these sidesets directly with This is in our workscope but for now, the user shouldn't really use any of these additionally generated sidesets beside the ones I mentioned in the previous comment

Reply:
It may be. But you may not knowingly add something that runs a bug. If you dont want to fix it, you ll have to remove this submeshgenerator from the workflow. There are other generators that can perform this you may try. But I'd rather you find the problem

Reply:
@GiudGiud after looking into this further, the reason for the strange sideset definitions is that there was a conflict with the sideset ID generated at the PinMeshGenerator/SidesetsFromNormalsGenerator (hard-coded as 1000x in the code), and the ones auto-generated by PCCMG (1000x and 1500x, as explained by @miaoyinb). I've temporarily hard coded the SidesetsFromNormalGenerator to be 1600x now, so the following sidesets are created in my current version:

1000x: auto-generated by PCCMG
1500x: auto-generated by PCCMG (the sum of 1000x and 1500x make the outer boundary)
1600x: defined after PinMeshGenerator/SidesetsFromNormalsGenerator (the sum of 1600x also make the outer boundary).

These are definitely redundant but defining the 1600x boundaries are required in order for PatternedMeshGenerator to be able to define outer boundaries correctly at the pin lattice definition stage in AssemblyMeshGenerator

Reply:
I have no context but I glanced onto this. Are these 1500+ boundaries needed in a production run?

Reply:
1000x and 1500x are not needed in the production run but 1600x is needed to define outer boundaries if the mesh generator is used in a lattice definition stage by an upper-level MG (e.g. AMG uses these pin boundaries to define the assembly lattice boundary, CMG uses these assembly boundaries to define the core lattice boundary). This will be much simpler and cleaner to define down the line once the underlying MG's used by RGMB can set these boundaries directly

Reply:
ok can we delete all these if they are not needed later? I think the 1600x are added in this PR. but if you could catch the 1000x and 1500x after they are last used, would be good too.

Having extra side sets in the output makes it less clear. And comes at a cost to ray tracing 

Use this
https://mooseframework.inl.gov/source/meshgenerators/BoundaryDeletionGenerator.html

Reply:
I've gone ahead and deleted the 1000x and 1500x sidesets, the source for PCCMG also had to be updated since there were certain instances where these sidesets weren't being generated by PCCMG so BoundaryDeletionGenerator would fail to find the sidesets to delete.

One note about the 1600x sidesets I had previously. I've removed this definition and used the external_boundary_name directly using SidesetsFromNormalsGenerator, but I also had to an additional RenameBoundaryGenerator to rename the sidesets because setting all of the boundary names to the same value in SidesetsFromNormalsGenerator would cause the external boundaries to not be created properly (see the part of the code where I name the sidesets tmp_left, tmp_right, ...). Of course this is not ideal, but we are planning to clean up the implementation of PatternedMeshGenerator (legacy mesh generator not initially in the reactor module workscope) so that it can generate outer boundaries directly without relying on all of these intermediate mesh generators, but this functionality and the resultant RGMB code cleanup will be part of a future PR.